tests/c_api/encode.cc

```diff
@@ -215,7 +215,7 @@ CASE("Encode from columnar data") {
 
     long icol[nrows] = {1111, 2222, 3333, 4444, 5555, 6666, 7777, 8888, 9999, 0};
     long bcol[nrows] = {1101, 2202, 3303, 4404,  5505, 6606, 7707, 8808, 9909, 0};
-    char scol[nrows][3 * sizeof(double)] = {0};
+    char scol[nrows][3 * sizeof(double)]{};
```

Comment:
These are not the same.

= {} should zero-initialise the memory. Please check if that makes a difference in this case. And if so, you need to provide a mechanism to do the zeroing.

Reply:
They behave identical with c++11 and later.

Both are aggregate initialization. `= {0}` is actually `={{0}}` due to brace elision and initializes `scol[0][0]` to `0`. Then copy list initialization is used for the remaining elements, which are value initialized.

`char[3][4]{}` is also an aggregate initialization with an empty init-list, so the same rule as above is applied for all elements: 

> Otherwise, if the initializer list has no elements, the object is value-initialized.

See also:
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf
11.6.1 / 11-12 pag2 226/227
11.6.4 / 3.10 page 234